client/orderbook/orderbook.go

```diff
@@ -558,3 +558,18 @@ func (ob *OrderBook) MidGap() (uint64, error) {
 	}
 	return (s[0].Rate + b[0].Rate) / 2, nil
 }
+
+// BestFill is the best (rate, quantity) fill for an order of the type and
+// quantity specified.
```

Comment:
Maybe mention that `BestFillMarketBuy` should be used for market buys, but this is good for limits and market sells, where quantity is in multiples of the base asset's lot size.